core/arch/arm/plat-imx6/platform_config.h

```diff
@@ -0,0 +1,192 @@
+ *  | TEETZ private RAM  |  TEE_RAM (SRAM)    |   ^
+ *  |                    +--------------------+   |
+ *  |                    |  TA_RAM            |   |
+ *  +-----------------------------------------+   | CFG_DDR_TEETZ_RESERVED_SIZE
+ *  |                    |      teecore alloc |   |
+ *  |  TEE/TZ and NSec   |  PUB_RAM   --------|   |
+ *  |   shared memory    |         NSec alloc |   |
+ *  +-----------------------------------------+   v
+ *
+ *  TEE_RAM : 256KByte
+ *  PUB_RAM : 1MByte
+ *  TA_RAM  : all what is left (at least 2MByte !)
+ */
+
+/* emulated SRAM, 256K at start of secure DDR */
+
+#define TZSRAM_BASE			    0x4E000000
+#define TZSRAM_SIZE			    0x00040000
+
+/* Location of trusted dram */
+
+#define CFG_DDR_TEETZ_RESERVED_START	0x4E100000
+#define CFG_DDR_TEETZ_RESERVED_SIZE	    0x01F00000
+
+#define CFG_PUB_RAM_SIZE		(1 * 1024 * 1024)
+
+#define TZDRAM_BASE			    (CFG_DDR_TEETZ_RESERVED_START)
+#define TZDRAM_SIZE			    (CFG_DDR_TEETZ_RESERVED_SIZE - \
+				CFG_PUB_RAM_SIZE)
+
+#define CFG_TA_RAM_START		TZDRAM_BASE
+#define CFG_TA_RAM_SIZE			TZDRAM_SIZE
+
+#define CFG_SHMEM_START			(CFG_DDR_TEETZ_RESERVED_START + \
+				TZDRAM_SIZE)
+#define CFG_SHMEM_SIZE			CFG_PUB_RAM_SIZE
+
+#define CFG_TEE_RAM_START		TZSRAM_BASE
+#define CFG_TEE_RAM_PH_SIZE		TZSRAM_SIZE
+
+#ifndef CFG_TEE_LOAD_ADDR
+#define CFG_TEE_LOAD_ADDR		TZSRAM_BASE
+#endif
+
+#else /* CFG_WITH_PAGER */
+
+/*
+ * TEE/TZ RAM layout:
+ *
+ *  +-----------------------------------------+  <- CFG_DDR_TEETZ_RESERVED_START
+ *  | TEETZ private RAM  |  TEE_RAM           |   ^
+ *  |                    +--------------------+   |
+ *  |                    |  TA_RAM            |   |
+ *  +-----------------------------------------+   | CFG_DDR_TEETZ_RESERVED_SIZE
+ *  |                    |      teecore alloc |   |
+ *  |  TEE/TZ and NSec   |  PUB_RAM   --------|   |
+ *  |   shared memory    |         NSec alloc |   |
+ *  +-----------------------------------------+   v
+ *
+ *  TEE_RAM : 1MByte
+ *  PUB_RAM : 1MByte
+ *  TA_RAM  : all what is left (at least 2MByte !)
+ */
+
+#define CFG_DDR_TEETZ_RESERVED_START	0x4E000000
+#define CFG_DDR_TEETZ_RESERVED_SIZE	    0x02000000
+
+#define CFG_PUB_RAM_SIZE		(1 * 1024 * 1024)
+#define CFG_TEE_RAM_PH_SIZE		(1 * 1024 * 1024)
+
+#define TZDRAM_BASE			    (CFG_DDR_TEETZ_RESERVED_START)
+#define TZDRAM_SIZE			    (CFG_DDR_TEETZ_RESERVED_SIZE - \
+				CFG_PUB_RAM_SIZE)
+
+#define CFG_TA_RAM_START		(CFG_DDR_TEETZ_RESERVED_START + \
+				CFG_TEE_RAM_PH_SIZE)
+#define CFG_TA_RAM_SIZE			(CFG_DDR_TEETZ_RESERVED_SIZE - \
+				CFG_TEE_RAM_PH_SIZE - \
+				CFG_PUB_RAM_SIZE)
+
+#define CFG_SHMEM_START			(CFG_DDR_TEETZ_RESERVED_START + \
+				TZDRAM_SIZE)
+#define CFG_SHMEM_SIZE			CFG_PUB_RAM_SIZE
+
+#define CFG_TEE_RAM_START		TZDRAM_BASE
+
+#ifndef CFG_TEE_LOAD_ADDR
+#define CFG_TEE_LOAD_ADDR		TZDRAM_BASE
+#endif
+
+#endif /* CFG_WITH_PAGER */
+
+/* UART1, shared with NS */
+
+#define DEVICE0_PA_BASE			ROUNDDOWN(UART1_BASE, \
```

Comment:
Please use the `register_phys_mem` macro instead of defining these `DEVICE*` macros